File: src/org/jgll/grammar/Opt.java
package org.jgll.grammar;

public class Opt extends Nonterminal {

	private static final long serialVersionUID = 1L;
	
	private Symbol s;
	
	public Opt(Symbol s) {
		super(s.getName() + "?");
		this.s = s;
	}
	
	public Symbol getSymbol() {
		return s;
	}

}

File: src-scala/org/jgll/grammar/Opt.scala
package org.jgll.grammar

import org.jgll.grammar.condition.Condition

@SerialVersionUID(1L)
class Opt(private var s: Symbol) extends Nonterminal(s.getName + "?") {

  def getSymbol(): Symbol = s

  override def addCondition(condition: Condition): Opt = {
    val opt = new Opt(this.s)
    opt.conditions ++= (this.conditions)
    opt.conditions += (condition)
    opt
  }
}
